File: 2021/07/28/main.go
package main

func beautifulArray(n int) []int {
	r := []int{1}
	for len(r) < n {
		tmp := make([]int, 0, len(r)*2)
		for _, v := range r {
			if v*2-1 <= n {
				tmp = append(tmp, v*2-1)
			}
		}
		for _, v := range r {
			if v*2 <= n {
				tmp = append(tmp, v*2)
			}
		}
		r = tmp
	}
	return r
}

File: 2021/07/28/main.py
class Solution:
    def beautifulArray(self, n: int) -> List[int]:
        r = [1]
        while len(r) < n:
            r = [i * 2 - 1 for i in r] + [i * 2 for i in r]
        return [i for i in r if i <= n]